core/src/scene/styleParam.cpp

```diff
@@ -456,9 +464,18 @@ bool StyleParam::isColor(StyleParamKey _key) {
 bool StyleParam::isWidth(StyleParamKey _key) {
     switch (_key) {
         case StyleParamKey::width:
-        case StyleParamKey::font_stroke_width:
         case StyleParamKey::outline_width:
         case StyleParamKey::size:
+        case StyleParamKey::font_stroke_width:
+            return true;
+        default:
+            return false;
+    }
+}
+
+bool StyleParam::is2DWidth(StyleParamKey _key) {
```

Comment:
I am not sure if `2DWidth` or `Widths2D` are appropriate names here. I can see it being confusing as its not exactly a width parameter! How about something more generic like, `2DValue` and `Values2D` respectively.

Apart from this all logic looks good here.